### PR TITLE
Update server_info/state w/ reporting mode details

### LINKED ISSUE #1446: server_info endpoint documentation error
In the documentation of the `server_info` endpoint, it's stated that `state_accounting.*.transitions` is Number, when in fact in the server response it is a String.

Additionally, the field `peers` seems to be either optional or deprecated/removed.

### PR DESCRIPTION
**Pull request by mDuo13**
- Update example responses from server_info
- Update data type of `state_accounting.*.transitions` field. (I am unclear when this field changed from number to string, but it did.)
- Add reporting mode specific fields to server_info docs
- Mark `peers` field as omitted from reporting mode
- Fix "external" gRPC link in server modes page
- Update "Configure gRPC" tutorial to better reflect current status of gRPC in rippled

This commit fixes #1446.

### CODE REVIEW

File: content/_snippets/etl-source-object.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+### ETL Source Object
+<!-- This nested object definition is identical across server_state and server_info -->
+
+On a reporting mode server, each member of the `etl_sources` field is an object with the following fields:
+
+| Field                       | Type    | Description |
+|-----------------------------|---------|-------------|
+| `connected`                 | Boolean | If `true`, the reporting mode server is connected to this p2p mode server. If `false`, that may indicate a network outage or that the p2p mode server is down. |
+| `grpc_port`                 | String  | The port number on the p2p mode server where this reporting mode server is configured to connect and retrieve ledger data via gRPC. |
```
Comment (amarantha-k) [inline content/_snippets/etl-source-object.md]: For consistency, change all "p2p" to "P2P".

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/server_state.md [hunk 2]
```diff
@@ -306,6 +310,11 @@ The `state` object may have some arrangement of the following fields:
 | `validation_quorum`              | Number          | Minimum number of trusted validations required to validate a ledger version. Some circumstances may cause the server to require more validations. |
 | `validator_list_expires`         | Number          | _(Admin only)_ When the current validator list will expire, in [seconds since the Ripple Epoch][], or 0 if the server has yet to load a published validator list. [New in: rippled 0.80.1][] |
 
+[Reporting mode]: rippled-server-modes.html
```
Comment (amarantha-k) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/server_state.md:313]: duplicate?

**Review by amarantha-k — APPROVED** [review]
Other than the two comments noted, the changes look good.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md [hunk 3]
```diff
@@ -276,11 +281,15 @@ The `info` object may have some arrangement of the following fields:
...
 | `pubkey_validator`                  | String          | _(Admin only)_ Public key used by this node to sign ledger validations. This _validation key pair_ is derived from the `[validator_token]` or `[validation_seed]` config field. |
+| `reporting`                         | Object          | _([Reporting mode](rippled-server-modes.html) servers only)_ Information about this server's reporting-mode specific configurations. |
+| `reporting.etl_sources`             | Array           | _([Reporting mode](rippled-server-modes.html) servers only)_ A list of p2p-mode servers this reporting mode is retrieving data from. Each entry in this array is an [ETL Source object](#etl-source-object). |
+| `reporting.is_writer`               | Boolean         | _([Reporting mode](rippled-server-modes.html) servers only)_ If `true`, this server is configured to write to the relational database with ledger data. If `false`, it is configured read-only and relies on other reporting mode servers to populate the database. |
```
Comment (cjcobb23) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md]: This is actually not just whether it is configured to write, but whether it actually is writing. There could be 10 servers configured to write, but only one will actually be writing at any given time.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md [hunk 4]
```diff
@@ -276,11 +281,15 @@ The `info` object may have some arrangement of the following fields:
...
+| `reporting.last_publish_time`       | String          | _([Reporting mode](rippled-server-modes.html) servers only)_ An ISO 8601 timestamp indicating when this server last saw a new validated ledger from any of its p2p mode sources. |
```
Comment (cjcobb23) [inline content/references/http-websocket-apis/public-api-methods/server-info-methods/server_info.md]: This is actually the last time the server published a ledger to its subscription streams. Doesn't have anything to do with p2p.

**Review by cjcobb23 — CHANGES_REQUESTED** [review]

File: content/_snippets/etl-source-object.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+### ETL Source Object
+<!-- This nested object definition is identical across server_state and server_info -->
+
+On a reporting mode server, each member of the `etl_sources` field is an object with the following fields:
+
+| Field                       | Type    | Description |
+|-----------------------------|---------|-------------|
+| `connected`                 | Boolean | If `true`, the reporting mode server is connected to this p2p mode server. If `false`, that may indicate a network outage or that the p2p mode server is down. |
```
Comment (cjcobb23) [inline content/_snippets/etl-source-object.md]: `false` indicates the reporting server can't connect to the p2p server for whatever reason. Could be down, the port could be specified incorrectly, could be getting rate limited, etc.

File: content/references/http-websocket-apis/public-api-methods/server-info-methods/server_state.md (hunk 2 above)
Comment (mDuo13) [reply content/references/http-websocket-apis/public-api-methods/server-info-methods/server_state.md:313]: Not a duplicate. These reference links are case-sensitive, and only apply in the file where they're defined (except when using includes, which is not the case for this one).

The uppercase-R version is used on 292-295 and the lowercase-r version is used on line 289.

**Review by cjcobb23 — APPROVED** [review]
LGTM

**Pull request merged**